api/graphql/types/reservation_unit/serializers.py

```diff
@@ -299,28 +314,19 @@ def update_hauki(instance: ReservationUnit) -> None:
                 raise GraphQLError("Sending reservation unit as resource to HAUKI failed.") from err
 
     @staticmethod
-    def handle_pricings(pricings: list[dict[Any, Any]], reservation_unit) -> None:
-        # Delete pricings that are not in the payload
-        if not ReservationUnitPricingHelper.contains_status(PricingStatus.PRICING_STATUS_ACTIVE, pricings):
-            ReservationUnitPricing.objects.filter(
-                reservation_unit=reservation_unit,
-                status=PricingStatus.PRICING_STATUS_ACTIVE,
-            ).delete()
-
-        if not ReservationUnitPricingHelper.contains_status(PricingStatus.PRICING_STATUS_FUTURE, pricings):
-            ReservationUnitPricing.objects.filter(
-                reservation_unit=reservation_unit,
-                status=PricingStatus.PRICING_STATUS_FUTURE,
+    def handle_pricings(pricings: list[dict[Any, Any]], reservation_unit: ReservationUnit) -> None:
+        with atomic():
+            # Delete pricings that are not in the payload
+            pricing_pks = [pricing.get("pk") for pricing in pricings if "pk" in pricing]
+            ReservationUnitPricing.objects.filter(reservation_unit=reservation_unit).exclude(
+                pk__in=pricing_pks
             ).delete()
```

Comment:
Doesn't this now delete past (inactive) pricings too?

Reply:
Yes, if they are not included in the pricings sent by the frontend.
Should I add a check here to prevent the deletion of any past pricings?

Reply:
That would be better since the frontend doesn't support transparent editing of the pricings and it is desirably to ensure that the inactive pricings are kept 'for the record'.